app/assets/javascripts/admin/resources/services/enterprises.js.coffee

```diff
@@ -38,3 +38,17 @@ angular.module("admin.resources").factory 'Enterprises', ($q, EnterpriseResource
 
     resetAttribute: (enterprise, attribute) ->
       enterprise[attribute] = @pristineByID[enterprise.id][attribute]
+
+    functionToPerformActionOnEnterpriseResource = (resourceAction) ->
+      (enterprise) ->
+        deferred = $q.defer()
+        resourceAction({id: enterprise.permalink}, ((data) =>
+          @pristineByID[enterprise.id] = angular.copy(data)
+          deferred.resolve(data)
+        ), ((response) ->
+          deferred.reject(response)
+        ))
+        deferred.promise
+
+    removeLogo: functionToPerformActionOnEnterpriseResource(EnterpriseResource.removeLogo)
```

Comment:
yes 👍  I'd call it performActionOnEnterpriseResource

and then something like
save: performActionOnEnterpriseResource(EnterpriseResource.$update) ?

Reply:
Haha! That function name does sound awkward. I wanted to distinguish between:

```javascript
return function() {
  performAction()
}
```

And:

```javascript
performAction()
```

I can update it.

About `EnterpriseResource.$update`, the existing `EnterpriseResource.$update` differs from `removeLogo` and `removePromoImage` in that the existing code doesn't actually use the enterprise object provided by the API endpoint. [A bit more about it here.](https://github.com/openfoodfoundation/openfoodnetwork/pull/2512#discussion_r212593094)

I'll look into it a bit more to see if it's safe to have `admin/enterprises#update` to return `Api::Admin::EnterpriseSerializer` instead of `Api::Admin::IndexEnterpriseSerializer`. Would you happen to know if there are possible authorization related issues?